src/mongodb/table.rs

```diff
@@ -0,0 +1,231 @@
+use crate::mongodb::connection_pool::MongoDBConnectionPool;
+use crate::mongodb::utils::expression::{combine_exprs_with_and, expr_to_mongo_filter};
+use crate::mongodb::Error;
+use async_trait::async_trait;
+use datafusion::common::project_schema;
+use datafusion::arrow::datatypes::SchemaRef;
+use datafusion::catalog::{Session, TableProvider};
+use datafusion::error::{DataFusionError, Result as DataFusionResult};
+use datafusion::execution::{TaskContext};
+use datafusion::logical_expr::{Expr, TableType};
+use datafusion::physical_expr::EquivalenceProperties;
+use datafusion::physical_plan::execution_plan::{Boundedness, EmissionType};
+use datafusion::physical_plan::stream::RecordBatchStreamAdapter;
+use datafusion::physical_plan::{
+    DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning, PlanProperties, SendableRecordBatchStream
+};
+use datafusion::sql::TableReference;
+use mongodb::bson::Document;
+use std::{any::Any, fmt, sync::Arc};
+use serde_json;
+use futures::TryStreamExt;
+
+#[derive(Debug)]
+pub struct MongoDBTable {
+    pool: Arc<MongoDBConnectionPool>,
+    schema: SchemaRef,
+    table_reference: Arc<TableReference>,
+}
+
+impl MongoDBTable {
+    pub async fn new(
+        pool: &Arc<MongoDBConnectionPool>,
+        table_reference: impl Into<TableReference>,
+    ) -> Result<Self, Error> {
+        
+        let table_reference = table_reference.into();
+        let schema= pool
+            .connect()
+            .await?
+            .get_schema(&table_reference)
+            .await?;
+
+        Ok(Self {
+            pool: Arc::clone(pool),
+            schema,
+            table_reference: Arc::new(table_reference),
+        })
+    }
+
+    
+}
+
+#[async_trait]
+impl TableProvider for MongoDBTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        Arc::clone(&self.schema)
+    }
+
+    fn table_type(&self) -> TableType {
+        TableType::Base
+    }
+
+    async fn scan(
```

Comment:
You will need to implement `supports_filters_pushdown` or else DataFusion will not push down any filters, and you will end up doing a full collection scan on every query. I assume this is a bug since you implement support for converting filters to MongoDB Documents.

Reply:
Great catch, thanks!